extensions/interactions/base_test.py

```diff
@@ -265,124 +265,74 @@ def test_html_field_types_to_rule_specs_mapping_are_valid(self):
         rule_descriptions_dict = json.loads(
             utils.get_file_contents(feconf.RULES_DESCRIPTIONS_FILE_PATH))
 
-
         # In the following part, we generate the html_field_types_to_rule_specs
         # dict based on the values in the rule_descriptions.json file.
         generated_html_field_types_dict = {}
-        html_types = (
-            [{
-                'type': 'DragAndDropHtmlString',
-                'format': 'string'
-            }, {
-                'type': 'ListOfSetsOfHtmlStrings',
-                'format': 'listOfSets'
-            }, {
-                'type': 'SetOfHtmlString',
-                'format': 'set'
-            }])
+        html_types = []
+        allowed_html_type_formats = ['listOfSets', 'set', 'string']
+        # Verify that each html_type dict has a format key, which must be
+        # unique. After verification we delete the key for comparison with
+        # the generated html_field_types_to_rule_specs dict. We compare
+        # everything except the format, because the format can't be generated
+        # from the rule_descriptions.json file.
+        for html_type, html_type_dict in (
+                html_field_types_to_rule_specs_dict.items()):
+            self.assertTrue('format' in html_type_dict)
+            self.assertTrue(
+                html_type_dict['format'] in allowed_html_type_formats)
+            del html_type_dict['format']
 
         for interaction_id, interaction_rules in (
                 rule_descriptions_dict.items()):
             for rule_type, rule_description in interaction_rules.items():
                 description = rule_description['description']
+                # Extract the html_types from the rule description.
+                html_types_in_description = (
+                    re.findall(r'{{[a-z]\|([^}]*)}', description))
+                for html_type in html_types_in_description:
```

Comment:
html_types = set([html_type in html_types_in_description if 'Html' in html_type])

Reply:
Here we need to append to an existing list (`html_types`), hence list comprehension will make it more unreadable.